app/graphql/mutations/delete_mutation.rb

```diff
@@ -0,0 +1,35 @@
+module Mutations
+  class DeleteMutation < BaseMutation
+    argument :id, ID, required: true
+
+    field :id, ID, 'The deleted object ID.', null: false
+
+    def resolve(id:)
+      object = ForemanGraphqlSchema.object_from_id(id, context)
+
+      unless object
+        raise GraphQL::ExecutionError.new(
+          _('Could not resolve ID.')
+        )
+      end
+
+      authorize!(object, :destroy)
+
+      validate_object(object, resource_class)
+
+      User.as(context[:current_user]) do
+        object.destroy!
+
+        unless object.destroyed?
+          raise GraphQL::ExecutionError.new(
+            _('Could not destroy the record.')
```

Comment:
Is there a way how to show nice errors telling us more about why we failed to destroy?

![destroy-fail](https://user-images.githubusercontent.com/2664090/54606371-f4b4c100-4a4b-11e9-846d-b43d21f42559.png)